src/app/authentication/authentication.component.ts

```diff
@@ -39,12 +41,20 @@ export class AuthenticationComponent extends GraphExplorerComponent {
 
   // Https://docs.microsoft.com/en-us/azure/active-directory/active-directory-v2-protocols-implicit
   public login() {
-    this.authService.login();
-    }
+    localStorage.setItem('status', 'authenticating');
+    this.changeDetectorRef.detectChanges();
+    this.authService.login()
+      .then((accessToken) => {
+        if (accessToken) {
+          this.displayUserProfile();
+          this.setPermissions();
```

Comment:
After we have acquired the token, we should set the status to authenticated to prevent the spinner from showing.
Then call changeDetector to trigger angular to run  a change detecting cycle.

```
if (accessToken) {
  ...
  localStorage.setItem('status', 'authenticated');
  this.changeDetectorRef.detectChanges();
}
```

Reply:
This will resolve the authenticating issues I found.